src/lib.rs

```diff
@@ -285,6 +289,78 @@ impl Into<Affine> for PublicKey {
     }
 }
 
+impl Serialize for PublicKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        serializer.serialize_str(&base64::encode(&self.serialize()[..]))
+    }
+}
+
+struct PublicKeyVisitor;
+
+impl<'de> de::Visitor<'de> for PublicKeyVisitor {
+    type Value = PublicKey;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str(
+            "a bytestring of either 33 (compressed), 64 (raw), or 65 bytes in length"
+        )
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+        where E: de::Error
+    {
+        let value: &[u8] = &base64::decode(value).unwrap();
+        match value.len() {
+            33 => {
+                let pkey = PublicKey::parse_slice(
+                    value,
+                    Some(PublicKeyFormat::Compressed),
+                );
+                if pkey.is_err() {
+                    Err(E::custom("Issue parsing compressed public key"))
+                } else {
+                    Ok(pkey.unwrap())
+                }
```

Comment:
The error handling here seems a bit more verbose than it needs to be. Would something like this work?

```rust
let value: &[u8] = &base64::decode(value).unwrap();
use PublicKeyFormat::*;
match value.len() {
  33 => PublicKey::parse_slice(value, Compressed)?,
  64 => PublicKey::parse_slice(value, Raw)?,
  65 => PublicKey::parse_slice(value, Full)?,
  len => E::custom(Error::InvalidInputLength),
}
```
…and then expand a bit on the crates `Error` type to provide a `Display` impl?